### ISSUE TITLE
Generate a ProjectTypeGuid property for known project types

### ISSUE THREAD

**Issue by jgoshi** [dotnet/sdk#522]
At least for c#, f#, and VB.

This will allow dotnet cli to figure out the project type guid when running dotnet add project for adding a project to a solution.

**Comment by davidfowl** [comment]
We did lots of work to avoid project type guids. Is this really required to add a project file to the solution? How does it work in the new project system today without a project type guid?

**Comment by jgoshi** [comment]
@srivatsn @davkean @dsplaisted @piotrpMSFT 
If I do a "dotnet add solution.sln project.ext" then in order to add the project to the solution I need some project type guid. It was recommended that I could use the CPS one {13B669BE-BB05-4DDF-9536-439F39A36129}. But it was also recommended that I should detect and insert the proper one for c#, F#, and vb. Unless others feel it's ok to just always use the CPS one?

**Comment by jgoshi** [comment]
@srivatsn @davkean @dsplaisted @piotrpMSFT Any comments? Should we scratch this work and dotnet add will always use the CPS project type guid in the sln file?

**Comment by srivatsn** [comment]
To answer @davidfowl's question Project Type Guids are how VS instantiates a project system for a project. It doesn't have to be in the project but it's in the solution. @jgoshi is not proposing to add it back to the project file itself but to set it in the SDK so that the CLI can query the project and figure out what guid to put in the solution when it adds the project. CPS has a project type guid as well.

**Comment by davkean** [comment]
@jgoshi I think we need to think about the correct factory to use here - if someone uses dotnet to add a .NET Framework traditional project, we do not want it to opt into CPS.

**Comment by srivatsn** [comment]
Yes the project type guid that's added by the CLI cannot always be the CPS guid. Hence the suggestion that the CLI should just query the project for the ProjectTypeGuid and that the SDK set it. For desktop projects that don't have a guid there's some logic that the solution uses to determine the guid right?

**Comment by DustinCampbell** [comment]
So, what's the plan here? At the moment, the CLI is always using the CPS guid, which is going to break if someone tries to add, say, a WinForms project to their solution with the CLI, right?

**Comment by srivatsn** [comment]
Daniel has a PR out to MSBuild to add the defaults but at this point that's post RTM. Do we think this meets the bar for RTM? Also didn't we say that the default guid was going to be the old C# GUID since that works for new projects as well?

**Comment by DustinCampbell** [comment]
I don't think it meets the bar. I just wanted to understand what the plan is since this bug didn't get updated with this info.

> Also didn't we say that the default guid was going to be the old C# GUID since that works for new projects as well?

That information isn't captured here either, so I couldn't tell you what you decided. :smile:

**Comment by jgoshi** [comment]
The current CLI behavior is to use the CPS guid. If that needs to change we'd need to do something real soon. @piotrpMSFT

**Comment by MiYanni** [comment]
There's a test, [WhenPassedAnUnknownProjectTypeItFails](https://github.com/dotnet/sdk/blob/941ec62ab28b01b4efcab6bcdaec37b2d319b965/test/dotnet-sln.Tests/GivenDotnetSlnAdd.cs#L977-L978), that is marked as Skip against this issue. The error it gives when ran:
```
Expected command to fail but it did not.
File Name: C:\Code\sdk\artifacts\bin\redist\Debug\dotnet\dotnet.exe
Arguments: sln App.sln add UnknownProject\UnknownProject.unknownproj
Exit Code: 0
StdOut:

StdErr:
Project 'C:\Code\sdk\artifacts\tmp\Debug\WhenPassedAnU---F3F8F9CE\UnknownProject\UnknownProject.unknownproj' has an unknown project type and cannot be added to the solution file. Contact your SDK provider for support.
```

I'm adjusting the StdError text in the test to match this as it seems appropriate. However, the test fails as it is expecting the `sln add` command to fail. But this situation only returns `0` as the exit code. So, it sounds like the intent is for the command itself to fail in this situation. I'm reopening this issue since this test is still failing.

**Comment by edvilme** [comment]
> There's a test, [WhenPassedAnUnknownProjectTypeItFails](https://github.com/dotnet/sdk/blob/941ec62ab28b01b4efcab6bcdaec37b2d319b965/test/dotnet-sln.Tests/GivenDotnetSlnAdd.cs#L977-L978), that is marked as Skip against this issue. The error it gives when ran:
> 
> ```
> Expected command to fail but it did not.
> File Name: C:\Code\sdk\artifacts\bin\redist\Debug\dotnet\dotnet.exe
> Arguments: sln App.sln add UnknownProject\UnknownProject.unknownproj
> Exit Code: 0
> StdOut:
> 
> StdErr:
> Project 'C:\Code\sdk\artifacts\tmp\Debug\WhenPassedAnU---F3F8F9CE\UnknownProject\UnknownProject.unknownproj' has an unknown project type and cannot be added to the solution file. Contact your SDK provider for support.
> ```
> 
> I'm adjusting the StdError text in the test to match this as it seems appropriate. However, the test fails as it is expecting the `sln add` command to fail. But this situation only returns `0` as the exit code. So, it sounds like the intent is for the command itself to fail in this situation. I'm reopening this issue since this test is still failing.

This was discussed briefly on #44570 and with vs-solutionpersistence team, and at the moment, a default guid is added.
